pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -446,19 +449,16 @@ func parsePerformanceProfileManifest(data []byte, nodePoolName string) (*perform
 	if !ok {
 		return nil, fmt.Errorf("error parsing PerformanceProfile manifests")
 	}
+	return performanceProfile, nil
+}
 
+func updatePerformanceProfileName(performanceProfile *performancev2.PerformanceProfile, nodePoolName string) {
 	// Make PerformanceProfile names unique if a PerformanceProfile is duplicated across NodePools
 	// for example, if one ConfigMap is referenced in multiple NodePools
-	performanceProfile.SetName(performanceProfile.ObjectMeta.Name + "-" + hashStruct(nodePoolName))
-	klog.V(2).Infof("parsePerformanceProfileManifest: name: %s", performanceProfile.GetName())
-
-	// Propagate NodePool name from ConfigMap down to PerformanceProfile object
-	if performanceProfile.Labels == nil {
-		performanceProfile.Labels = make(map[string]string)
+	if !strings.HasSuffix(performanceProfile.ObjectMeta.Name, "-"+hashStruct(nodePoolName)) {
+		performanceProfile.SetName(performanceProfile.ObjectMeta.Name + "-" + hashStruct(nodePoolName))
+		klog.V(2).Infof("updatePerformanceProfile: name: %s", performanceProfile.GetName())
```

Comment:
this and the other log (line 159) seems redundant. Maybe consolidate in one? (and let's always log this rename)